client/agent_endpoint.go

```diff
@@ -0,0 +1,167 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/command/agent/monitor"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/ugorji/go/codec"
+
+	metrics "github.com/armon/go-metrics"
+	log "github.com/hashicorp/go-hclog"
+	sframer "github.com/hashicorp/nomad/client/lib/streamframer"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+)
+
+type Agent struct {
+	c *Client
+}
+
+func NewAgentEndpoint(c *Client) *Agent {
+	m := &Agent{c: c}
+	m.c.streamingRpcs.Register("Agent.Monitor", m.monitor)
+	return m
+}
+
+func (m *Agent) monitor(conn io.ReadWriteCloser) {
+	defer metrics.MeasureSince([]string{"client", "monitor", "monitor"}, time.Now())
```

Comment:
just a question as to why this was done under `client/monitor/monitor` instead of `client/agent/monitor` ?

Reply:
Ah thanks good catch, this was changed from monitor_endpoint to agent_endpoint and this part was missed, will update